registry_module.go

```diff
@@ -45,14 +45,26 @@ type RegistryModules interface {
 	ReadVersion(ctx context.Context, moduleID RegistryModuleID, version string) (*RegistryModuleVersion, error)
 
 	// Delete a registry module
+	// Deprecated: Use DeleteByName
 	Delete(ctx context.Context, organization string, name string) error
 
 	// Delete a specific registry module provider
+	// Deprecated: Use DeleteByProvider
 	DeleteProvider(ctx context.Context, moduleID RegistryModuleID) error
 
 	// Delete a specific registry module version
+	// Deprecated: Use DeleteByVersion
 	DeleteVersion(ctx context.Context, moduleID RegistryModuleID, version string) error
 
+	// Delete a registry module by name
+	DeleteByName(ctx context.Context, module RegistryModuleDeleteByNameOrProvider) error
+
+	// Delete a specific registry module by provider
+	DeleteByProvider(ctx context.Context, module RegistryModuleDeleteByNameOrProvider, provider string) error
```

Comment:
DeleteProvider and DeleteVersion already use RegistryModuleID as a parameter, so it doesn't appear that we are adding any information to the API by adding the new methods. Can we just replace the implementation to use the new endpoints and use the data found in RegistryModuleID to request them?

Reply:
Validation on `RegistryModuleID` is not the same as validation on `RegistryModuleDeleteByNameOrProvider`. I had to build a new struct to accommodate the API calls requiring parameters that weren't originally required. Maybe there's a simpler implementation?

Reply:
You could have two different validation methods on RegistryModuleID --but let's resolve the other conversation first.